[LonelyFort] iP

LonelyFort opened the pull request:
# Angela

A straightforward and convenient task and reminder app which you can count on to keep track of the never-ending responsibilities in your life.

It's:
- [x] Ad-free
- [x] Free of charge
- [x] Non-subscription based (surprisingly hard to come by)
- [x] Clear and straightforward text-commands
- [x] Runs on Java (Because developing applications on JavaScript is for cowards) 
- [x] Open Source 
- [x] No need to install third party library  
- [x] Has nice art when you boot up the app

## Features (Ver 1.2)
- Manage Todo, Deadline and Events capable of handling dates.
- Load and store tasks onto local device.
- Filter tasks that contains specific keywords. ~~(In Progress)~~ **(Available in v1.1)**
- Graphic User Interface for users.

## Installation Instructions
1. If you have not installed Java 17 onto your local device, you may download it from [here](https://www.oracle.com/java/technologies/javase/jdk17-archive-downloads.html).
2. Download latest release of Angela.jar from [here](https://github.com/LonelyFort/ip/releases).
3. Place the jar file into an empty folder (Preferably under the same name Angela).

## Run Instructions
4. Once the file is placed, open the terminal by right click -> **Open In Terminal**.
5. Enter the following into the terminal window to run the program: `java -jar .\Angela.jar`

## Disclaimer
As a new file will be created when the program is running, please ensure you do not set the root file to require admin permission for the program to run smoothly.

## Random Quote Because the Requirement Says So (And a random Emoji too)

> "That's that and this is this. 😏 " - Some washed-up Grade 9 Fixer.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LonelyFort committed:
Rename Duke to Athena and implement greet

LonelyFort committed:
Add echo functionality to Athena

LonelyFort committed:
adding list function to Athena

LonelyFort committed:
Rename Athena to Angela and modify responses to inputs

LonelyFort committed:
change array to arraylist and added task class

LonelyFort committed:
Implemented functionality of Angela to handle Todo Deadline and Event subtasks

LonelyFort committed:
added two different tests for Angela and added temporary error handling for check and uncheck functions

LonelyFort committed:
Added exception handling to Angela

LonelyFort committed:
Added delete functionality to Angela

LonelyFort committed:
added Javadoc to Angela

LonelyFort committed:
Added javadoc to all files and ensure all packages and classes following cs2103t coding standard

LonelyFort committed:
Changed Deadline and Event class to accept LocalDate instead of String

LonelyFort committed:
Added javadoc for utility class DateTimeValueHandler and InvalidDateException

LonelyFort committed:
make small changes for level 8 increment

LonelyFort committed:
implement the save functionality to Angela

LonelyFort committed:
fix READMe typo

LonelyFort committed:
Merge branch 'branch-Level-7'

LonelyFort committed:
add a new validation to check if end date is earlier than start date

LonelyFort committed:
add a new validation method to test if "to" date is earlier than "from" date for event tasks

LonelyFort committed:
fixed exception import to fit java coding standard

LonelyFort committed:
small changes on this branch to create a merge commit

LonelyFort committed:
Merge branch 'branch-Level-8'

LonelyFort committed:
separate UI and Command  of Angela as individual classes

LonelyFort committed:
git ignore saved tasks stored in local folder

LonelyFort committed:
Merge branch 'branch-A-MoreOOP'

LonelyFort committed:
Merge branch 'add-gradle-support'

LonelyFort committed:
implement gradle to project

LonelyFort committed:
add JUnit test to Angela

LonelyFort committed:
Merge branch 'branch-A-JUnit'

LonelyFort committed:
set up environments for jar file creation

LonelyFort committed:
Merge branch 'branch-A-Jar'

LonelyFort committed:
add javadocs to all classes in Angela

LonelyFort committed:
Merge branch 'branch-A-JavaDoc'

LonelyFort committed:
implement the find function for Angela

LonelyFort committed:
Merge branch 'branch-Level-9'

LonelyFort committed:
minor fixes for Angela and add funcrtionality to handle empty inputs

LonelyFort committed:
Merge branch 'branch-A-CodingStandard'

Reshiro commented on src/main/java/Angela/Angela.java:
```diff
@@ -0,0 +1,190 @@
+        setTimeout(() -> {
+            this.listData = new TaskList();
+            this.database = new Database(filePath, listData);
+        },10000);
```
```suggestion
        }, 10000);
```
Commas should be followed by a white space.

Reshiro commented on src/main/java/Angela/Angela.java:
```diff
@@ -0,0 +1,190 @@
+        while (true) {
+            String input = scan.nextLine().strip();
+            // ignores comments and empty lines
+            if (input.isEmpty() || input.charAt(0) == '/') continue;
```
We're required to always use braces for if statements unfortunately.

AnWe11 replied:
Why is there a need to check for '/'? Is it in case the user forgets the command word?

Reshiro commented on src/main/java/Angela/Angela.java:
```diff
@@ -0,0 +1,190 @@
+
+    //@@author Oleg Mikhailov-reused
+    //Reused from https://stackoverflow.com/questions/26311470/what-is-the-equivalent-of-javascript-settimeout-in-java
+    public static void setTimeout(Runnable runnable, int delay){
```
```suggestion
    public static void setTimeout(Runnable runnable, int delay) {
```
To be honest I'm not sure about the proper procedures for editing reused code, but if it's possible then a space would be a good idea here.

Reshiro commented on src/main/java/Angela/Angela.java:
```diff
@@ -0,0 +1,190 @@
+                Thread.sleep(delay);
+                runnable.run();
+            }
+            catch (Exception e){
```
```suggestion
            catch (Exception e) {
```

Reshiro commented on src/main/java/Angela/Angela.java:
```diff
@@ -0,0 +1,190 @@
+    //@@author
+
+    /**
+     * Main method for Angela.
+     * @param args
```
Remember to add an empty line before the parameters section, but I think that the param annotation here is not needed at all.

Reshiro commented on src/main/java/Angela/UI/UI.java:
```diff
@@ -0,0 +1,50 @@
+/**
+ * UI class to handle display of responses in the console.
+ */
+public class UI {
```
I think the proper naming is Ui? But I think it's best to look at the acronym naming guidelines because it's not that clear.

AnWe11 commented on src/main/java/Angela/Angela.java:
```diff
@@ -0,0 +1,190 @@
+                            @@.@//////////////@&&&&&@&&&&&&&&&&&@/////@ #                   (@         \s
+                            @@ ////////////////@&&&&&&&&&&&&&&&&@//////% %                   @@@       \s
+                           @@(#/////////////////&&&&&@&&&&&&&&&&%///////& *                   @@@      \s
+                           @  //////////////////(&&&&@&&&&&&&&&&@////////@ .                   @@      \s
+                          @@(%///////////////////@&&&&&&&&&&&&&&@/////////@                     @@     \s
+                          @  /////////////////////&&&&@&&&&&&&&&&//////////#                    @@     \s
+                        @@@.(//////////////////////@&&&&&&&&&&&&&(///////////  .                 @@#   \s
+                        @@  ////////////////////////@&&&&&&&&&&&&@///////////*( #                 @#   \s
+                        @@( /////////////////////////&&@&&&&&&&&&&@@@@@@@@@@@%/@                  @@@  \s
+                        @@ #////////////////////////@@&@&&&&&&&&&@@@           @@                  @@  \s
+                        @@ /////////////////////@@@ @@@@&&&&&&&&&@              @@@ %              @@@ \s
+                        @@ //////////////////@@       @@&&&&&&&&&@                @@.              ,@@ \s
+                        @@,///////////////@@@         @@&&&&&&&&@@                  @@ .            @@ \s
+                        @@%////////////@@             @@&&&&&&&&@                   @@@ *           &@@\s
+                        @@@//////////@@@              @@&&&&&&&@@                    *@@             @@\s
+                          @////////%@                 @@&&&&&&&@@                      @#           *@ \s
+                          @@//////@@                  @@&&&&&&@@                        @.         @@@ \s
+                          *@////@@@                   @@&&&&&&@@                        @@        @%*  \s
+                           @@//@@                     @@&&&&&&@@                         @@ *   @@#/   \s
+                           @@&@@                      @@&&&&&&@@                         @@@ #@@@      \s
+                            @@@                       @@&&&&&&@@                           @@@@@       \s
+                                                    @@&&&&&&&&@@                           @@          \s
+                                                  @@&&&&&&&&&&@@                                       \s
+                                                 @@@@#@&&&&&&&@@                                       \s
+                                                 @####&&&&&&&&@@                                       \s
+                                                  @@@@&&&&&&&&@@                                       \s
+                                                    @@&&&&&&&&%@                                       \s
+                                                    @##@&&&&@##@                                       \s
+                                                    @@########@@                                       \s
+                                                      @@####@@@                                        \s
+                        
+                        
+                Welcome, Manager. I am your assistant AI, Angela.
+                I am here to support you as it's your first day. 
+                I will provide practical advice and emotional support until you get used to your work.
+                
+                What is your request?                        
+                """
+        ), 7000);
+    }
```
Amazing ASCII art!

But wouldn't this be better placed in the UI class?

AnWe11 commented on src/main/java/Angela/exceptions/taskcreation/DateOrderException.java:
```diff
@@ -0,0 +1,19 @@
+     */
+    @Override
+    public String toString() {
+        return "End date and time cannot be earlier than the start date and time. We are not in a time loop.";
```
What if we are... D:

Reshiro commented on src/main/java/Angela/Angela.java:
```diff
@@ -0,0 +1,190 @@
+     * ASCII art depicts Angela. from Lobotomy Corp.
+     */
+    private void greet() {
+        setTimeout(() -> System.out.println("Initiating..."), 1000);
```
I like the use of delays for the welcome message.

Reshiro commented on src/main/java/Angela/storage/Database.java:
```diff
@@ -0,0 +1,104 @@
+package Angela.storage;
+
+import java.io.*;
```
IntelliJ might have added a wildcard import here automatically.

Reshiro commented on src/main/java/Angela/tasktype/Task.java:
```diff
@@ -0,0 +1,91 @@
+     *
+     * @return True if the task is completed, otherwise false.
+     */
+    public boolean isCompleted() {
```
Should the method name be more like a verb here?

AnWe11 commented on src/main/java/Angela/storage/TaskList.java:
```diff
@@ -0,0 +1,118 @@
+     *
+     * @return a string representation of all tasks in the list, formatted for saving
+     */
+    public String saveAllTask() {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < size(); i++) {
+            String taskToBeSaved = get(i).toSaveFormat();
+            if (i < size() - 1) {
+                taskToBeSaved += "\n";
+            }
+            stringBuilder.append(taskToBeSaved);
+        }
+        return stringBuilder.toString();
+    }
```
This is an extremely elegant way to save the tasks, good job!

Reshiro reviewed (commented):
The code is very high quality, and I like the Lobotomy Corp reference. There's just a few minor deviations from the recommended coding standard.

AnWe11 commented on src/main/java/Angela/UI/UI.java:
```diff
@@ -0,0 +1,50 @@
+package Angela.UI;
+
+/**
+ * UI class to handle display of responses in the console.
```
Although your explanation for your comments are crystal clear, maybe your comments can be more succinct?

AnWe11 commented on src/main/java/Angela/UI/UI.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public class UI {
+
+    /**
+     * Displays response in a simple 2 line UI.
+     *
+     * @param response The string to be displayed.
+     */
+    public static void displayResponse(String response) {
+        System.out.println("________________________________________________________\n"
+                + response + "\n"
+                + "________________________________________________________");
+    }
```
Is there a need for a documentation here for a simple function?

AnWe11 reviewed (commented):
I think your code is extremely organised and I have a lot to learn from you. I like the way you packaged and compartmentalised your classes and even for your exceptions. The way you read from the textfile to update the taskList was also innovative. Overall, great job on this and Angela will definitely be a hit.

LonelyFort committed:
fix minor cosmetic issue

Co-authored-by: Michael Cheong <michaelcheong7@gmail.com>

LonelyFort committed:
fix minor code quality issues

LonelyFort committed:
Merge branch 'master' of https://github.com/LonelyFort/ip

LonelyFort committed:
fix code quality issues based on checkstyle

LonelyFort committed:
implement a simple GUI for Angela

LonelyFort committed:
Merge branch 'branch-Level-10'

LonelyFort committed:
add javadoc to new components

LonelyFort committed:
update buid.gradle

LonelyFort committed:
Change code to adhere to code quality in Week 5's topics, and fix a
critical bug regarding find function in Angela.

In Week 5's code quality topics, it is recommended to add a default
branch for case statements, handlePrintTask and
handleTaskModification fail this code quality check.

For the find function bug, I have taken notice that Angela will
throw IndexOutOfBound exception when entering the command "find "
(with a space). This has been fixed.

LonelyFort committed:
add Assertions to Command.java and add banner and icon to GUI of
Angela

Assertions are added to handle command functions to terminate the
program when a command is erroneously sent to the function. This
add another layer of guard from unexpected behaviours.

Angela GUI will now be displayed as Angela in the window banner,
and Lobotomy Corp Icon will be displayed as window Icon.

LonelyFort committed:
fix code quality in accordance to week 5 standards

LonelyFort committed:
implement streams into Angela and fix critical bug in delete
function in Angela

LonelyFort committed:
Merge branch 'branch-A-Assertions'

LonelyFort committed:
s

LonelyFort committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/Angela/app/App.java
#	src/main/java/Angela/util/Command.java

LonelyFort committed:
Merge branch 'master' into branch-A-Streams

LonelyFort committed:
Merge branch 'branch-A-CodeQuality'

LonelyFort committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/Angela/util/Command.java

LonelyFort committed:
Merge branch 'branch-A-Streams'

LonelyFort committed:
fix small issue that occurred while merging

LonelyFort committed:
small change for PR

LonelyFort committed:
Merge pull request #1 from LonelyFort/branch-A-Assertions

small change for PR

LonelyFort committed:
create small change for PR

LonelyFort committed:
Merge pull request #2 from LonelyFort/branch-A-CodeQuality

create small change for PR

LonelyFort committed:
make small change for PR

LonelyFort committed:
Merge pull request #3 from LonelyFort/branch-A-Streams

make small change for PR

LonelyFort committed:
set up CI using gradle.yml

LonelyFort committed:
make small fixes to Angela

LonelyFort committed:
implemented importance tag to tasks and add simplification to
commands

LonelyFort committed:
fix some bugs that occurred when implementing the importance
and simplicity function

LonelyFort committed:
Update gradle.yml

LonelyFort committed:
update test files

LonelyFort committed:
Merge pull request #4 from LonelyFort/branch-B-Tagging&Simplification

add extension to Angela

LonelyFort committed:
fix some typos and update outdated javadocs

LonelyFort committed:
change angela package name in accordance to coding standard and fix some coding quality issues

LonelyFort committed:
update build.gradle file for changed package name

LonelyFort committed:
Merge pull request #6 from LonelyFort/branch-more-code-quality

change angela package name in accordance to coding standard and fix s…

LonelyFort committed:
Update README.md

LonelyFort committed:
Update README.md

LonelyFort committed:
update READ.md in docs file and add Ui.png

LonelyFort committed:
Merge branch 'master' of https://github.com/LonelyFort/ip

LonelyFort committed:
Update README.md

LonelyFort committed:
Update README.md

LonelyFort committed:
Update README.md

LonelyFort committed:
improve savedTaskParser error handling and add a manual command
for link to user manual

LonelyFort committed:
change manual command response

LonelyFort committed:
Update README.md

LonelyFort committed:
Update README.md

LonelyFort committed:
update deadline and event command to prevent creation of tasks with
empty descriptions

LonelyFort committed:
Merge branch 'master' of https://github.com/LonelyFort/ip

LonelyFort committed:
revert savedTaskParser changes and add back
UnreadableFileException
